### ISSUE TITLE
Unable to ‘map g’

### ISSUE THREAD

**Issue by mina86** [vimperator/vimperator-labs#252]
I’m trying to use ‘g’ to switch to the next tab. In desperation, this is what I used:

    noremap g :tn<CR>
    nnoremap g :tn<CR>
    vnoremap g :tn<CR>

Unfortunately, ‘g’ is still recognized as a prefix only so on its own it does nothing, but ‘gg’ scrolls to the top. The same issue is with ‘z’ which is also a prefix key.

**Comment by gkatsev** [comment]
This is also an issue in vim itself. I think the only way to map just `g` is to basically map all the other things to `<Nop>`.

**Comment by mina86** [comment]
Is there some quick way to grep all the bindings that start with g? Or do I
have to go through :help looking for them, hoping I won't miss any?

**Comment by gkatsev** [comment]
`:usage mappings`

**Comment by mina86** [comment]
No cigar:

    map g$ <Nop>
    map g0 <Nop>
    map g<C-g> <Nop>
    map g<lt> <Nop>
    map g@ <Nop>
    map gB <Nop>
    map gF <Nop>
    map gH <Nop>
    map gP <Nop>
    map gT <Nop>
    map gU <Nop>
    map gb <Nop>
    map gf <Nop>
    map gg <Nop>
    map gh <Nop>
    map gi <Nop>
    map gn <Nop>
    map go <Nop>
    map gt <Nop>
    map gu <Nop>
    map zI <Nop>
    map zM <Nop>
    map zO <Nop>
    map zR <Nop>
    map zZ <Nop>
    map zi <Nop>
    map zm <Nop>
    map zo <Nop>
    map zr <Nop>
    map zz <Nop>
    noremap z :undo<CR>
    noremap g :tn<CR>

still doesn’t work.

**Comment by gkatsev** [comment]
Thanks for trying. It's possible that it's mapping specially. In which case a change to the code would be necessary beforehand.

**Comment by timss** [comment]
As a reference of sorts:

If a map is not executed, chances are that longer maps starting with the same key sequence still exist. 

Take a look at [events.js' onKeyPress logic](https://github.com/vimperator/vimperator-labs/blob/b7effe239e8f9459bdd0a6c577096380467b636d/common/content/events.js#L1028).

Using explicit internal functions:

    :js mappings.add([modes.NORMAL], ["g"], "foobar", function() { alert("foobar") });

If a simple `g` is used, nothing happens, however `g<Esc>` will execute the function.

Same goes for mappings with keys that are not used anywhere else, such as `|`:

    :js mappings.add([modes.NORMAL], ["|"], "foobar", function() { alert("foobar") });

Now `|` will execute immediately because there is no other mappings to be resolved.

---

I don't think adding lots of maps to `<Nop>` will not help. Nor does [`unm[ap]`](https://github.com/vimperator/vimperator-labs/blob/144894bd8cc79998c6c0e7a022374ef982e3d727/common/content/mappings.js#L538) seem to do the trick.

If you really want to use `g` for anything, you may have to remove mappings from the code itself. And there's lots of them, and they come in many [shapes](https://github.com/vimperator/vimperator-labs/blob/66e29f8c1895e6e399c62e533accc03226249d25/common/content/tabs.js#L1040). 

**tl;dr** bad idea.

---

Closing, as this issue doesn't have a "solution" with the current design.

**Comment by gkatsev** [comment]
If someone can go through and figure out how to make vimperator actually acknowledge a full set of `<Nop>` so that the key could be mapped without waiting for the timeout to trigger, we would happily merge such a PR but it is unlikely that any of the "core" developers are going to take any of this on because of our limited constraints on just trying to keep vimperator afloat.
